apps/android/app/src/main/java/chat/simplex/app/views/newchat/QRCodeScanner.kt

```diff
@@ -36,74 +37,86 @@ fun QRCodeScanner(onBarcode: (String) -> Unit) {
         )
         implementationMode = PreviewView.ImplementationMode.COMPATIBLE
       }
-    },
-//    modifier = Modifier.fillMaxSize(),
-    update = { previewView ->
-      val cameraSelector: CameraSelector = CameraSelector.Builder()
-        .requireLensFacing(CameraSelector.LENS_FACING_BACK)
-        .build()
-      val cameraExecutor: ExecutorService = Executors.newSingleThreadExecutor()
-      val cameraProviderFuture: ListenableFuture<ProcessCameraProvider> =
-        ProcessCameraProvider.getInstance(context)
+    }
+  ) { previewView ->
+    val cameraSelector: CameraSelector = CameraSelector.Builder()
+      .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+      .build()
+    val cameraExecutor: ExecutorService = Executors.newSingleThreadExecutor()
+    val cameraProviderFuture: ListenableFuture<ProcessCameraProvider> =
+      ProcessCameraProvider.getInstance(context)
 
-      cameraProviderFuture.addListener({
-        preview = Preview.Builder().build().also {
-          it.setSurfaceProvider(previewView.surfaceProvider)
-        }
-        val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
-        val barcodeAnalyser = BarCodeAnalyser { barcodes ->
-          barcodes.firstOrNull()?.rawValue?.let(onBarcode)
+    cameraProviderFuture.addListener({
+      preview = Preview.Builder().build().also {
+        it.setSurfaceProvider(previewView.surfaceProvider)
+      }
+      val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
+      val qrReader = QRCodeReader()
+      fun getQR(imageProxy: ImageProxy) {
+        val currentTimeStamp = System.currentTimeMillis()
+        if (currentTimeStamp - lastAnalyzedTimeStamp >= TimeUnit.SECONDS.toMillis(1)) {
+          lastAnalyzedTimeStamp = currentTimeStamp
+          val bitmap = imageProxyToBinaryBitmap(imageProxy)
+          val decodeHints = EnumMap<DecodeHintType, Any>(
+            DecodeHintType::class.java
+          )
+          decodeHints[DecodeHintType.POSSIBLE_FORMATS] = BarcodeFormat.QR_CODE
+          try {
+            val result = qrReader.decode(bitmap, decodeHints)
+            onBarcode(result.text)
+            imageProxy.close()
+          } catch (nfe: NotFoundException) {
+            // Comment log for too much noise
+            // Log.d(TAG, "Failed to find QR code in current image.")
+            imageProxy.close()
+          } catch (fe: FormatException) {
+            Log.d(TAG, "Failed to find QR code of expected format. Possible mis-detection.")
+            imageProxy.close()
+          } catch (ce: ChecksumException) {
+            Log.d(TAG, "Invalid QR Code. Checksum failure.")
+            imageProxy.close()
+          } catch (e: Exception) {
+            Log.d(TAG, "QR code detection failed for unknown reason: ${e.localizedMessage}")
+            imageProxy.close()
```

Comment:
Logs useful for debugging, could collapse or at least remove logging for production. Not too bad now as logging at 1hz in worst case.